addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2102,11 +2109,20 @@ options.registry.topMenuColor = options.Class.extend({
      */
     selectStyle(previewMode, widgetValue, params) {
         this._super(...arguments);
-        const className = widgetValue ? (params.colorPrefix + widgetValue) : '';
-        this.trigger_up('action_demand', {
-            actionName: 'toggle_page_option',
-            params: [{name: 'header_color', value: className}],
-        });
+        if (widgetValue && !ColorpickerWidget.isCSSColor(widgetValue)) {
+            widgetValue = params.colorPrefix + widgetValue;
+        }
+        if (params.topMenuProperty === 'text-color'){
+            this.trigger_up('action_demand', {
+                actionName: 'toggle_page_option',
+                params: [{name: 'header_text_color', value: widgetValue}],
+            });
+        } else {
+            this.trigger_up('action_demand', {
+                actionName: 'toggle_page_option',
+                params: [{name: 'header_color', value: widgetValue}],
+            });
+        }
```

Comment:
use a variable for params.name to avoid to duplicate the trigger_up ;)

Reply:
Done! 👍